kustomization/components/paperless/statefulset.yml

```diff
@@ -0,0 +1,102 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: paperless
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: paperless
+  serviceName: paperless
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: paperless
+    spec:
+      volumes:
+        - name: paperless-data
+          persistentVolumeClaim:
+            claimName: paperless-data
+        - name: paperless-media
+          persistentVolumeClaim:
+            claimName: paperless-media
+        - name: paperless-scans
+          persistentVolumeClaim:
+            claimName: paperless-scans
+        - name: paperless-export
+          persistentVolumeClaim:
+            claimName: paperless-export
+      initContainers:
+        # Paperless needs the trash dir to exist on startup
+        - name: prep-trash-dir
+          image: ghcr.io/paperless-ngx/paperless-ngx
+          command:
+            - /bin/sh
+            - -c
+            - mkdir -p $PAPERLESS_TRASH_DIR
+          env:
+            - name: PAPERLESS_TRASH_DIR
+              value: "/usr/src/paperless/media/trash"
+          volumeMounts:
+            - mountPath: "/usr/src/paperless/media"
+              name: paperless-media
+      containers:
+        - name: paperless
+          image: ghcr.io/paperless-ngx/paperless-ngx
+          ports:
+            - containerPort: 8000
+          env:
+            - name: PAPERLESS_PORT
+              value: "8000"
+            - name: PAPERLESS_REDIS
+              value: "redis://redis:6379"
```

Comment:
It uses a namespace-relative address to the redis instance in the component - seemed safe given we don't have a solution to renaming when using components and there's no reason to change the default redis port that I know of.

Reply:
Where did we settling on naming standards?  Are we suffixing things (so this would be `redis-svc` then)?  I have a preference for this because it makes things more grep-able in a non-ambiguous way.

Reply:
Yeah, improving the grep-ability seems good here since I don't know of any tools with a semantic understanding here.